etcdserver/api/v3rpc/util.go

```diff
@@ -81,3 +81,9 @@ func togRPCError(err error) error {
 	}
 	return grpcErr
 }
+
+func isClientCtxErr(err error) bool {
+	ev, _ := status.FromError(err)
```

Comment:
just to be more defensive, we should check the error against context.DeadlineExceeded and context.Canceled when the fromError fails with error. i am not sure if grpc.send and recv will always return statusError type

Reply:
Sure. Does it make sense that we pass ```stream.Context().Err()``` (or  ```sws.gRPCStream.Context().Err()``` when needed) to this func and modify the func as below?:
```
func isClientCtxErr(err error) bool {
	if err == nil {
		return false
	}
	return err == context.Canceled || err == context.DeadlineExceeded
}
```

Reply:
i am not sure how exactly the client side context error gets propagated to server side. maybe the stream.Context().Err() only returns the server side context error not the one propagated from the client.

Reply:
Thanks @xiang90 I see. I was able to run manual tests with using stream.Context().Err() but I will dig more. @gyuho if you have any suggestions please let me know. Thanks!!

Reply:
Yeah maybe add another parameter to `isClientCtxErr` function to parse `stream.Context` errors as well?

Reply:
Thanks @gyuho !!